static/app/views/starfish/components/sampleEvents.tsx

```diff
@@ -0,0 +1,100 @@
+import Duration from 'sentry/components/duration';
+import GridEditable, {GridColumnHeader} from 'sentry/components/gridEditable';
+import Link from 'sentry/components/links/link';
+import EventView from 'sentry/utils/discover/eventView';
+import {
+  DiscoverQueryProps,
+  useGenericDiscoverQuery,
+} from 'sentry/utils/discover/genericDiscoverQuery';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import {TextAlignLeft} from 'sentry/views/starfish/modules/APIModule/endpointTable';
+
+type Props = {
+  eventView: EventView;
+};
+
+type DataRow = {
+  id: string;
+  'transaction.duration': number;
+};
+
+const COLUMN_ORDER = [
+  {
+    key: 'id',
+    name: 'Event ID',
+    width: 300,
+  },
+  {
+    key: 'transaction.duration',
+    name: 'Duration',
+    width: -1,
+  },
+];
+
+export function SampleEvents({eventView}: Props) {
+  const location = useLocation();
+  const organization = useOrganization();
+
+  const sampleEventsEventView = eventView
+    .clone()
+    .withColumns([
+      {
+        field: 'transaction.duration',
+        kind: 'field',
+      },
+    ])
+    .withSorts([
+      {
+        field: 'transaction.duration',
+        kind: 'desc',
+      },
+    ]);
+
+  function renderBodyCell(column: GridColumnHeader, row: DataRow): React.ReactNode {
+    if (column.key === 'id') {
+      return (
+        <Link to={`/performance/${row['project.name']}:${row.id}`}>
+          {row.id.slice(0, 8)}
+        </Link>
+      );
+    }
+
+    if (column.key === 'transaction.duration') {
+      return (
+        <Duration
+          seconds={row['transaction.duration'] / 1000}
+          fixedDigits={2}
+          abbreviation
+        />
+      );
+    }
+
+    return <TextAlignLeft>{row[column.key]}</TextAlignLeft>;
+  }
+
+  const {isLoading, data} = useGenericDiscoverQuery<any, DiscoverQueryProps>({
+    route: 'events',
+    eventView,
+    referrer: 'starfish-transaction-summary-sample-events',
+    limit: 5,
+    location,
+    orgSlug: organization.slug,
+    getRequestPayload: () => ({
+      ...sampleEventsEventView.getEventsAPIPayload(location),
+    }),
+  });
```

Comment:
Could you clarify the reason why we are passing in the `eventView` here for the first parameters, but using the request payload from the `sampleEventsEventView`?

Could we use the `useDiscoverQuery` hook, and just pass in the `sampleEventsEventView`?

Reply:
Typo, fixed